transcendence-app/src/user/user.controller.spec.ts

```diff
@@ -47,24 +56,46 @@ describe('User Controller unit tests', () => {
     controller = module.get<UserController>(UserController);
   });
 
-  it('creates a user', async () => {
-    const user = await controller.addUser(testUserDto);
-    expect(user.id).toEqual(testUserId);
-    expect(user.username).toEqual(testUserDto.username);
-  });
+  describe('addUser', () => {
+    describe('when user with username does not exists', () => {
+      it('should create a new user', async () => {
+        mockUserService.retrieveUserWithUserName = () => Promise.resolve(null);
+        const user = await controller.addUser(testUserDto);
+        expect(user.id).toEqual(testUserId);
+        expect(user.username).toEqual(testUserDto.username);
+      });
+    });
 
-  it('returns an existing user', async () => {
-    const user = await controller.getUserById(testUserId);
-    expect(user.id).toEqual(testUserId);
+    describe('when user with username exists', () => {
```

Comment:
I think the choice of not allowing duplicated usernames is slightly questionable, but I don't care enough to stop the merge.

Reply:
We could discuss these options at the next meeting. I'll add a point in Notion.